File: backend/app/schemas/content.py
from typing import Literal

from pydantic import BaseModel, Field

HeroTextPosition = Literal["left", "center", "right"]
# "split": mitad imagen / mitad panel de texto. "full": imagen a lo ancho
# completo, con el texto (si hay) superpuesto según text_position.
HeroLayout = Literal["split", "full"]


class HeroSlide(BaseModel):
    id: str
    image_desktop: str
    image_mobile: str | None = None
    html: str = ""
    layout: HeroLayout = "split"
    text_position: HeroTextPosition = "left"
    text_color: str = "#1A1A1A"
    bg_color: str = "#F5F5F3"
    bg_opacity: int = Field(100, ge=0, le=100)
    link_url: str | None = None
    active: bool = True
    alt: str = ""


class HeroSlideInput(BaseModel):
    id: str | None = None
    image_desktop: str
    image_mobile: str | None = None
    html: str = ""
    layout: HeroLayout = "split"
    text_position: HeroTextPosition = "left"
    text_color: str = "#1A1A1A"
    bg_color: str = "#F5F5F3"
    bg_opacity: int = Field(100, ge=0, le=100)
    link_url: str | None = None
    active: bool = True
    alt: str = ""


class HeroSettings(BaseModel):
    slides: list[HeroSlide]


class HeroSettingsUpdate(BaseModel):
    slides: list[HeroSlideInput] = Field(default_factory=list)


class AboutChapter(BaseModel):
    id: str
    eyebrow: str = ""
    title: str = ""
    text: str = ""
    image: str = ""


class AboutChapterInput(BaseModel):
    id: str | None = None
    eyebrow: str = ""
    title: str = ""
    text: str = ""
    image: str = ""


class AboutSettings(BaseModel):
    hero_image: str = ""
    hero_title: str = ""
    intro_title: str = ""
    intro_text: str = ""
    chapters: list[AboutChapter] = Field(default_factory=list)
    gallery: list[str] = Field(default_factory=list)


class AboutSettingsUpdate(BaseModel):
    hero_image: str = ""
    hero_title: str = ""
    intro_title: str = ""
    intro_text: str = ""
    chapters: list[AboutChapterInput] = Field(default_factory=list)
    gallery: list[str] = Field(default_factory=list)


class InspirationProject(BaseModel):
    id: str
    title: str = ""
    description: str = ""
    location: str = ""
    image: str = ""


class InspirationProjectInput(BaseModel):
    id: str | None = None
    title: str = ""
    description: str = ""
    location: str = ""
    image: str = ""


class InspirationSettings(BaseModel):
    hero_image: str = ""
    hero_title: str = ""
    intro_title: str = ""
    intro_text: str = ""
    projects: list[InspirationProject] = Field(default_factory=list)


class InspirationSettingsUpdate(BaseModel):
    hero_image: str = ""
    hero_title: str = ""
    intro_title: str = ""
    intro_text: str = ""
    projects: list[InspirationProjectInput] = Field(default_factory=list)


class DesignProject(BaseModel):
    id: str
    title: str = ""
    description: str = ""
    location: str = ""
    image: str = ""


class DesignProjectInput(BaseModel):
    id: str | None = None
    title: str = ""
    description: str = ""
    location: str = ""
    image: str = ""


class DesignSettings(BaseModel):
    hero_image: str = ""
    hero_title: str = ""
    intro_title: str = ""
    intro_text: str = ""
    projects: list[DesignProject] = Field(default_factory=list)


class DesignSettingsUpdate(BaseModel):
    hero_image: str = ""
    hero_title: str = ""
    intro_title: str = ""
    intro_text: str = ""
    projects: list[DesignProjectInput] = Field(default_factory=list)


# Páginas de solo texto del footer (Envíos, Medios de pago, Cambios y
# devoluciones, Preguntas frecuentes, Términos y condiciones): un título de
# página y una lista de secciones cortas (subtítulo + párrafo).
class InfoPageSection(BaseModel):
    id: str
    title: str = ""
    text: str = ""


class InfoPageSectionInput(BaseModel):
    id: str | None = None
    title: str = ""
    text: str = ""


class InfoPageSettings(BaseModel):
    title: str = ""
    sections: list[InfoPageSection] = Field(default_factory=list)


class InfoPageSettingsUpdate(BaseModel):
    title: str = ""
    sections: list[InfoPageSectionInput] = Field(default_factory=list)


# Links a redes sociales que se muestran en el footer. El vendedor elige la
# red (define el ícono) y pega la URL.
SocialPlatform = Literal[
    "instagram",
    "facebook",
    "pinterest",
    "x",
    "youtube",
    "tiktok",
    "linkedin",
    "threads",
    "whatsapp",
]


class SocialLink(BaseModel):
    id: str
    platform: SocialPlatform
    url: str = ""


class SocialLinkInput(BaseModel):
    id: str | None = None
    platform: SocialPlatform
    url: str = ""


class SocialSettings(BaseModel):
    links: list[SocialLink] = Field(default_factory=list)


class SocialSettingsUpdate(BaseModel):
    links: list[SocialLinkInput] = Field(default_factory=list)

File: frontend/src/types/content.ts
export type HeroTextPosition = "left" | "center" | "right";

/** "split": mitad imagen / mitad panel de texto. "full": imagen a lo ancho
 * completo, con el texto (si hay) superpuesto según text_position. */
export type HeroLayout = "split" | "full";

export interface HeroSlide {
  id: string;
  image_desktop: string;
  image_mobile: string | null;
  html: string;
  layout: HeroLayout;
  text_position: HeroTextPosition;
  /** Color del texto (hex), aplicado vía la variable CSS --hero-text. */
  text_color: string;
  /** Color de fondo del panel de texto (hex). */
  bg_color: string;
  /** Opacidad del fondo del panel, 0-100. */
  bg_opacity: number;
  link_url: string | null;
  active: boolean;
  alt: string;
}

export interface HeroSettings {
  slides: HeroSlide[];
}

export interface AboutChapter {
  id: string;
  eyebrow: string;
  title: string;
  text: string;
  image: string;
}

export interface AboutSettings {
  hero_image: string;
  hero_title: string;
  intro_title: string;
  intro_text: string;
  chapters: AboutChapter[];
  gallery: string[];
}

export interface InspirationProject {
  id: string;
  title: string;
  description: string;
  location: string;
  image: string;
}

export interface InspirationSettings {
  hero_image: string;
  hero_title: string;
  intro_title: string;
  intro_text: string;
  projects: InspirationProject[];
}

export interface DesignProject {
  id: string;
  title: string;
  description: string;
  location: string;
  image: string;
}

export interface DesignSettings {
  hero_image: string;
  hero_title: string;
  intro_title: string;
  intro_text: string;
  projects: DesignProject[];
}

/** Páginas de solo texto del footer: Envíos, Medios de pago, Cambios y
 * devoluciones, Preguntas frecuentes, Términos y condiciones. */
export type InfoPageSlug =
  | "envios"
  | "medios-de-pago"
  | "cambios-y-devoluciones"
  | "preguntas-frecuentes"
  | "terminos-y-condiciones";

export interface InfoPageSection {
  id: string;
  title: string;
  text: string;
}

export interface InfoPageSettings {
  title: string;
  sections: InfoPageSection[];
}

/** Redes sociales del footer. La `platform` define el ícono y el orden en
 * el que se listan es el orden en que se muestran. */
export type SocialPlatform =
  | "instagram"
  | "facebook"
  | "pinterest"
  | "x"
  | "youtube"
  | "tiktok"
  | "linkedin"
  | "threads"
  | "whatsapp";

export interface SocialLink {
  id: string;
  platform: SocialPlatform;
  url: string;
}

export interface SocialSettings {
  links: SocialLink[];
}
